clickhouse_search/models.py

```diff
@@ -69,6 +72,249 @@ def __init__(self, name, select='*', order_by=None):
+            for population, field in self.model.POPULATION_FIELDS
+        }
+
+    def _filter_frequency(self, results, freqs=None, pathogenicity=None, **kwargs):
+        frequencies =  freqs or {}
+        clinvar_override_q = self._clinvar_path_q(pathogenicity)
+
+        for population, pop_filter in frequencies.items():
+            pop_subfields = self.populations.get(population)
+            if not pop_subfields:
+                continue
+
+            if pop_filter.get('af') is not None and pop_filter['af'] < 1:
+                af_field = next(field for field in ['filter_af', 'af'] if field in pop_subfields)
+                if af_field:
+                    af_q = Q(**{
+                        f'populations__{population}__{af_field}__lte': pop_filter['af'],
+                    })
+                    if clinvar_override_q and pop_filter['af'] < PATH_FREQ_OVERRIDE_CUTOFF:
+                        af_q |= clinvar_override_q
+                    results = results.filter(af_q)
+            elif pop_filter.get('ac') is not None:
+                results = results.filter(**{f'populations__{population}__ac__lte': pop_filter['ac']})
+
+            if pop_filter.get('hh') is not None:
+                for subfield in ['hom', 'hemi']:
+                    if subfield not in pop_subfields:
+                        continue
+                    hh_q = Q(**{
+                        f'populations__{population}__{subfield}__lte': pop_filter['hh'],
+                    })
+                    if clinvar_override_q:
+                        hh_q |= clinvar_override_q
+                    results = results.filter(hh_q)
+
+        return results
+
+    def _filter_in_silico(self, results, in_silico=None, **kwargs):
+        allowed_scores = {score for score, _ in self.model.PREDICTION_FIELDS}
+        in_silico_filters = {
+            score: value for score, value in (in_silico or {}).items() if score in allowed_scores and value
+        }
+        if not in_silico_filters:
+            return results
+
+        in_silico_q = None
+        for score, value in in_silico_filters.items():
+            score_q = self._get_in_silico_score_q(score, value)
+            if in_silico_q is None:
+                in_silico_q = score_q
+            else:
+                in_silico_q |= score_q
+
+        if not in_silico.get('requireScore', False):
+            in_silico_q |= Q(**{f'predictions__{score}__isnull': True for score in in_silico_filters.keys()})
+
+        return results.filter(in_silico_q)
+
+    @staticmethod
+    def _get_in_silico_score_q(score, value):
+        score_column = f'predictions__{score}'
+        try:
+            return Q(**{f'{score_column}__gte': float(value)})
+        except ValueError:
+            return Q(**{score_column: value})
+
+    @classmethod
+    def _filter_annotations(cls, results, annotations=None, pathogenicity=None, exclude=None, gene_ids=None, **kwargs):
```

Comment:
this is copy-paste from the old entries manager, the only change is to remove the `key__` prefix from the filter fields